discovery_child_development/utils/huggingface_pipeline.py

```diff
@@ -0,0 +1,292 @@
+from datasets import Dataset
+import torch
+import transformers
+import evaluate
+from transformers import (
+    EvalPrediction,
+    AutoTokenizer,
+    AutoModelForSequenceClassification,
+    TrainingArguments,
+    Trainer,
+    EarlyStoppingCallback,
+)
+from transformers.models.distilbert.tokenization_distilbert_fast import (
+    DistilBertTokenizerFast,
+)
+from transformers.models.distilbert.modeling_distilbert import (
+    DistilBertForSequenceClassification,
+)
+from nesta_ds_utils.loading_saving import S3
+from discovery_child_development import S3_BUCKET
+from discovery_child_development.utils.general_utils import make_tarfile
+from typing import Union
+from pathlib import Path
+import numpy as np
+import pandas as pd
+from sklearn.metrics import f1_score, roc_auc_score, accuracy_score
+
+
+def create_labels(dataset: Dataset, cols_to_skip: list) -> Dataset:
+    """Add labels to the dataset. Labels are a list of 0.0s and 1.0s
+    corresponding to the order of the labels in the original dataframe
+    that was used to create the dataset. Note the 0.0s and 1.0s must be floats."""
+    cols = dataset.column_names
+    return dataset.map(
+        lambda row: {
+            "labels": torch.FloatTensor(
+                [(row[col]) for col in cols if col not in cols_to_skip]
+            )
+        }
+    )
+
+
+def load_tokenizer(config: dict, problem_type: bool = True) -> DistilBertTokenizerFast:
+    """Load multi label classification BERT tokenzier
+
+    Args:
+        config: Dictionary containing model config
+
+    Returns:
+        BERT tokenizer
+    """
+    if problem_type:
+        return AutoTokenizer.from_pretrained(
+            "distilbert-base-uncased", problem_type=config["problem_type"]
```

Comment:
It specifically isn't so it takes all the arguments from a config file (e.g. binary_classifier/taxonomy_classifier) so that you could define which problem_type you would want up top.

Reply:
Yep sorry, just looked at this again and I can see that the first argument in that function is `config`! So indeed, just me being obtuse 🤦🏻‍♀️